jupyter-js-widgets/src/widget_bool.js

```diff
@@ -60,71 +66,70 @@ var CheckboxView = widget.DOMWidgetView.extend({
          * model to update.
          */
         var value = this.model.get('value');
-        this.model.set('value', ! value, {updated_view: this});
+        this.model.set('value', !value, {updated_view: this});
         this.touch();
     },
 
-    update: function(options){
+    update: function(options) {
         /**
          * Update the contents of this view
          *
          * Called when the model is changed. The model may have been
          * changed by another view or by a state update from the back-end.
          */
-        this.$checkbox.prop('checked', this.model.get('value'));
+        this.checkbox.checked = this.model.get('value');
 
         if (options === undefined || options.updated_view != this) {
-            this.$checkbox.prop("disabled", this.model.get("disabled"));
+            this.checkbox.disabled = this.model.get('disabled');
 
-            var description = this.model.get("description");
+            var description = this.model.get('description');
             if (description.trim().length === 0) {
-                this.$label.hide();
+                this.label.style.display = 'none';
             } else {
-                this.typeset(this.$label, description);
-                this.$label.show();
+                this.typeset(this.label, description);
+                this.label.style.display = '';
             }
         }
         return CheckboxView.__super__.update.apply(this);
-    },
+    }
 });
 
 var ToggleButtonModel = BoolModel.extend({
     defaults: _.extend({}, BoolModel.prototype.defaults, {
-        _view_name: "ToggleButtonView",
-        _model_name: "ToggleButtonModel",
-        tooltip: "",
-        icon: "",
-        button_style: "",
-    }),
+        _view_name: 'ToggleButtonView',
+        _model_name: 'ToggleButtonModel',
+        tooltip: '',
+        icon: '',
+        button_style: ''
+    })
 });
 
 var ToggleButtonView = widget.DOMWidgetView.extend({
+    initialize: function() {
+        /**
+         * Called when view is instantiated.
+         */
+        this.setElement(document.createElement('button'));
+        ToggleButtonView.__super__.initialize.apply(this, arguments);
+    },
+
     render: function() {
         /**
          * Called when view is rendered.
          */
-        var that = this;
-        this.setElement($('<button />')
-            .addClass('jupyter-widgets widget-toggle-button btn btn-default')
-            .attr('type', 'button')
-            .on('click', function (e) {
-                e.preventDefault();
-                that.handle_click();
-            }));
-        this.$el.attr("data-toggle", "tooltip");
-        this.listenTo(this.model, "change:button_style", this.update_button_style, this);
+        this.el.className = 'jupyter-widgets widget-toggle-button';
```

Comment:
A widget view object can be created (i.e., that is where setElement should be called), and then the container can assign classes to the element, and then render the widget. This may be theoretical at this point (especially since setElement was in the render up to this point), but using className would break things in this case. For example, it would probably be cleaner for the generic jupyter-widgets class to be applied by the widget manager.